Automattic/jetpack
# Update eslint-plugin-wpcalypso

#### Changes proposed in this Pull Request:

Calypso has discontinued their eslint-config-wpcalypso package, merging
it into eslint-plugin-wpcalypso.

The new version also brings some rule changes:

* `react/jsx-uses-react` is now disabled
* `react/react-in-jsx-scope` is now disabled
* `wpcalypso/import-docblock` no longer exists. Hooray, no more silly
  comment blocks! See also p4TIVU-9O9-p2.
* New rule `wpcalypso/no-unsafe-wp-apis`, which IMO is a good idea. I've
  added ignores for the existing hits.

We should probably follow Calypso's lead and replace
`wpcalypso/import-docblock` with `import/order`, but as that will update
tons of files I'm going to leave that for a followup.

#### Jetpack product discussion


None

#### Does this pull request change what data or activity we track or use?


No

#### Testing instructions:






* Is CI happy?
* There should be no changes to the build output due to this change.
